src/instruments/src/PFD/index.jsx

```diff
@@ -0,0 +1,204 @@
+    getSimVar, setSimVar, renderTarget, createDeltaTimeCalculator,
+} from '../util.mjs';
+import { SmoothSin, LagFilter, RateLimiter } from './PFDUtils.jsx';
+import './style.scss';
+
+const AltModeSelected = () => {
+    if (Simplane.getAutoPilotAltitudeManaged() && getSimVar('L:AP_CURRENT_TARGET_ALTITUDE_IS_CONSTRAINT', 'number') !== 0) {
+        return false;
+    }
+    return true;
+};
+
+// eslint-disable-next-line react/prefer-stateless-function
+class PFD extends Component {
+    constructor(props) {
+        super(props);
+
+        const url = document.getElementsByTagName('a32nx-pfd-element')[0].getAttribute('url');
+        this.dispIndex = parseInt(url.substring(url.length - 1), 10);
+
+        this.deltaTime = 0;
+        this.GetDeltaTime = createDeltaTimeCalculator();
+        this.prevAirspeed = 0;
+
+        this.VLs = 0;
+        this.VAlphaProt = 0;
+        this.VAlphaLim = 0;
+        this.VS = 0;
+
+        this.barTimer = 10;
+
+        this.smoothFactor = 0.5;
+
+        this.isAttExcessive = false;
+
+        this.AirspeedAccFilter = new LagFilter(1.2);
+        this.AirspeedAccRateLimiter = new RateLimiter(1.2, -1.2);
+
+        this.LSButtonPressed = false;
+
+        // for testing only
+        setSimVar('L:A32NX_MachPreselVal', -1, 'mach');
+        setSimVar('L:A32NX_SpeedPreselVal', -1, 'knots');
+    }
+
+    componentDidMount() {
+        renderTarget.parentElement.addEventListener('update', () => {
+            this.update(this.GetDeltaTime());
+        });
+        renderTarget.parentElement.addEventListener(`A320_Neo_PFD_BTN_LS_${this.dispIndex}`, () => {
+            this.onLSButtonPressed();
+        });
+    }
+
+    onLSButtonPressed() {
+        this.LSButtonPressed = !this.LSButtonPressed;
+        setSimVar(`L:BTN_LS_${this.dispIndex}_FILTER_ACTIVE`, this.LSButtonPressed, 'Bool');
+    }
+
+    smoothSpeeds(_dTime, _vls, _vaprot, _valim, _vs) {
+        const seconds = _dTime / 1000;
+        this.VLs = SmoothSin(this.VLs, _vls, this.smoothFactor, seconds);
+        this.VAlphaProt = SmoothSin(this.VAlphaProt, _vaprot, this.smoothFactor, seconds);
+        this.VAlphaLim = SmoothSin(this.VAlphaLim, _valim, this.smoothFactor, seconds);
+        this.VS = SmoothSin(this.VS, _vs, this.smoothFactor, seconds);
+    }
+
+    update(_deltaTime) {
+        this.deltaTime = _deltaTime;
+        this.forceUpdate();
+    }
+
+    render() {
+        const pitch = -getSimVar('PLANE PITCH DEGREES', 'degrees');
+        const roll = getSimVar('PLANE BANK DEGREES', 'degrees');
+        const heading = getSimVar('PLANE HEADING DEGREES MAGNETIC', 'degrees');
+
+        if (!this.isAttExcessive && (pitch > 25 || pitch < -13 || Math.abs(roll) > 45)) {
+            this.isAttExcessive = true;
+        } else if (this.isAttExcessive && pitch < 22 && pitch > -10 && Math.abs(roll) < 40) {
+            this.isAttExcessive = false;
+        }
+
+        const groundTrack = getSimVar('GPS GROUND MAGNETIC TRACK', 'degrees');
```

Comment:
As this seems to report weird values near standstill I would suggest maybe just overriding this with the magnetic heading below a certain airspeed. E.g. compare https://www.youtube.com/watch?v=n3TJMro_NzA#t=1m28s (especially the pushback with a "backwards" ground track) with https://discord.com/channels/738864299392630914/739223625898655854/809194933247672360. In the reference footage, during takeoff, the two only seem to diverge starting at 50-60 knots.